src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -121906,6 +121908,80 @@ void M()
                 );
         }
 
+        [Fact, WorkItem(41437, "https://github.com/dotnet/roslyn/issues/41437")]
```

Comment:
@dotnet/roslyn-compiler This is fixing a bug we noticed in LDM last week with a `TNotNull`. 

Note that fixing this makes it harder to use `TryGet` APIs (with `[MaybeNullWhen(...)]` with a `TNotNull`) and this problem hit the roslyn codebase. 
@cston I wonder if the introduction of `T??` later in C# 9 might improve that situation (ie. by inferring `TNotNull??` for the `out var` instead of `TNotNull`). Something to consider...

Reply:
@jaredpar From discussion with Chuck, it's probably better to hold this PR back and ship it along with `T??` work. I'll stop by to talk.

Reply:
From offline discussion with Jared, we'll hold this PR until we do the `T??` work.